### PR TITLE
kube-aggregator APIServiceRegistrationController remove watch services 

### LINKED ISSUE #57836: kube-aggregator APIServiceRegistrationController no need watch services
**Is this a BUG REPORT or FEATURE REQUEST?**:

 /kind bug

**What happened**:

Currently APIServiceRegistrationController watch service change and find the related APIServices related if any. Then it AddAPIService/RemoveAPIService for APIAggregator.  

I dig into the kube-aggregator and find it no need totally, because `AvailableConditionController` update 
APIServices status when related service or endpoint change. So if  `APIServiceRegistrationController ` does not watch services, it will not miss any event for  APIServices.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:

>Currently APIServiceRegistrationController watch service change and find the related APIServices related if any. Then it AddAPIService/RemoveAPIService for APIAggregator.

>I dig into the kube-aggregator and find it no need totally, because AvailableConditionController update
APIServices status when related service or endpoint change. So if APIServiceRegistrationController does not watch services, it will not miss any event for APIServices.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #57836

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go [hunk 1]
```diff
@@ -164,10 +148,16 @@ func (c *APIServiceRegistrationController) addAPIService(obj interface{}) {
 	c.enqueue(castObj)
 }
 
-func (c *APIServiceRegistrationController) updateAPIService(obj, _ interface{}) {
-	castObj := obj.(*apiregistration.APIService)
-	glog.V(4).Infof("Updating %s", castObj.Name)
-	c.enqueue(castObj)
+func (c *APIServiceRegistrationController) updateAPIService(old, new interface{}) {
+	oldObj := old.(*apiregistration.APIService)
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go]: revert this change, resync is a backstop to catch any inconsistencies that creep in, and doesn't happen frequently

File: staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go [hunk 2]
```diff
@@ -314,10 +314,16 @@ func (c *AvailableConditionController) addAPIService(obj interface{}) {
 	c.enqueue(castObj)
 }
 
-func (c *AvailableConditionController) updateAPIService(obj, _ interface{}) {
-	castObj := obj.(*apiregistration.APIService)
-	glog.V(4).Infof("Updating %s", castObj.Name)
-	c.enqueue(castObj)
+func (c *AvailableConditionController) updateAPIService(old, new interface{}) {
+	oldObj := old.(*apiregistration.APIService)
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go]: revert this change as well

File: staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go [hunk 3]
```diff
@@ -368,8 +374,15 @@ func (c *AvailableConditionController) addService(obj interface{}) {
 	}
 }
 
-func (c *AvailableConditionController) updateService(obj, _ interface{}) {
-	for _, apiService := range c.getAPIServicesFor(obj.(*v1.Service)) {
+func (c *AvailableConditionController) updateService(old, new interface{}) {
+	oldObj := old.(*v1.Service)
```
Comment (liggitt) [inline staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go]: and this

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go]: make sense. But I doubt why some other controllers compare pod ResourceVersion .

File: staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go (hunk 2 above)
Comment (hzxuzhonghu) [reply staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go]: ok

File: staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go (hunk 1 above)
Comment (hzxuzhonghu) [reply staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go]: After some deep re-thought and look into the code, I think if APIService periodic resync with same resource version, the sync loop will make a dry-run, since it get `APIService` from local cache. Which still can not `catch any inconsistencies that creep in`.

**Comment by hzxuzhonghu** [comment]
@liggitt reverted

**Comment by liggitt** [comment]
/lgtm
/assign @deads2k
for approval

**Comment by hzxuzhonghu** [comment]
@liggitt Thanks for so quick response.
